packages/block-library/src/navigation-link/edit.js

```diff
@@ -105,6 +117,39 @@ function NavigationLinkEdit( {
 		}
 	}, [ url ] );
 
+	useEffect( () => {
```

Comment:
It'd be great to be able to wrap this up into a hook called something like `useIsDraggingWithin` (though no need to extract it into a separate file just yet):
```jsx
const elementRef = useRef();
const isDraggingWithin = useIsDraggingWithin( elementRef );

return ( 
  <SomeElement
    ref={ elementRef }
    className={ classnames( { 'is-dragging-within': isDraggingWithin } ) }
  />
);
```

The reason is, I think it'll be useful for the List View feature if we add collapsing/expanding to that at some point, like the mock ups in this issue - https://github.com/WordPress/gutenberg/issues/24029#issue-659717213

Reply:
Hmm, although I realise after writing this that in list view the collapsed section will likely expand when dragging in, but then remain open afterwards.

Reply:
Always a good idea to extract some logics into its own function though 👍